### PR TITLE
[Android Auto] App for testing stable versions

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

tl;dr changes
- Make a new `android-auto-app` for testing android auto specifically
- Remove android auto from `qa-test-app`
- Remove `debugApi project(":libnavigation-android")` in favor of `api("com.mapbox.navigation:android:2.9.0")`

This will unblock this upgrade https://github.com/mapbox/mapbox-navigation-android/pull/6588

#### More description

There are new changes that are part of the 2.10.0-alpha that are breaking experiences. There are also api conflicts when building between different versions, and it makes it very difficult to test "what is and what isn't" part of the android auto library.

We're releasing stable apis soon, so it's important to be able to test them. So we need a test app that is specific to androidauto. Unfortunately these issues also mean we need to remove androidauto from `qa-test-app`, unless we decide on a new solution (considering it's not ideal, a new solution could be taken soon).

This project is a copy paste of the example app https://github.com/mapbox/mapbox-navigation-android-examples/tree/main/android-auto-app

One of the only differences is, the android auto is a `project` instead of "versioned" dependency. Also the name of the app is different.
``` groovy
dependencies {
    // Mapbox Navigation Android Auto SDK
    implementation project(':libnavui-androidauto')

    // This library will depend on the latest stable version.
    implementation("com.mapbox.navigation:ui-dropin:2.9.0")

    // Dependencies needed for this example.
    implementation dependenciesList.androidXAppCompat
}
```

### CODE REVIEW

File: libnavui-androidauto/build.gradle [hunk 1]
```diff
@@ -40,12 +40,11 @@ dependencies {
...
     // This defines the minimum version of Navigation which is included in this SDK. To upgrade the
     // Navigation versions, you can specify a newer version in your downstream build.gradle.
-    releaseApi("com.mapbox.navigation:android:2.9.0-rc.2")
-    debugApi project(":libnavigation-android")
+    api("com.mapbox.navigation:android:2.9.1")
 
     // Search is currently in beta so it is not included in this SDK. The functionality of search
     // is included behind this library's api.
-    implementation(dependenciesList.mapboxSearchSdk)
+    implementation("com.mapbox.search:mapbox-search-android:1.0.0-beta.38.1")
```
Comment (Zayankovsky) [inline libnavui-androidauto/build.gradle:47]: I think this should be reverted. `1.0.0-beta.38.1` uses incompatible version of Common SDK. 
Also `mapboxSearchSdk` is not used anywhere else, we can safely update it, if necessary for Android Auto.

Comment (kmadsen) [reply libnavui-androidauto/build.gradle:47]: The libnavui-androidauto sdk will be pinned to a separate version of search. Also bumping it causes a crash https://github.com/mapbox/mapbox-navigation-android/pull/6588#issuecomment-1312292125

File: android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import kotlinx.coroutines.launch
+
+/**
+ * This is a temporarily solution for syncing two new libraries, Drop-in-ui and the Mapbox AA.
+ *
+ * The libraries are defining public apis so that there can be options to determine the experience
+ * while both the car and phone are displayed.
+ */
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class CarAppSyncComponent private constructor() : MapboxNavigationObserver {
```
Comment (Zayankovsky) [inline android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt:31]: Found the following scenario when it doesn't work:
1. Launch Android Auto and go to route preview. 
2. Launch the app on the phone. 
3. Android Auto unexpectedly transitions to Free Drive state.

File: libnavui-androidauto/build.gradle (hunk 1 above)
Comment (Zayankovsky) [reply libnavui-androidauto/build.gradle:47]: We can remove the constant then. Also it is really strange that you run into this crash. It doesn't happen in 1TAP, where we have been using that version for a long time.

**Review by Zayankovsky — APPROVED** [review]

File: libnavui-androidauto/build.gradle (hunk 1 above)
Comment (kmadsen) [reply libnavui-androidauto/build.gradle]: updated to use 39 again. that crash must be local to my machine. also 39 should be the [better version to use](https://github.com/mapbox/mapbox-search-android/compare/v1.0.0-beta.38.1...v1.0.0-beta.39#diff-6f145b4e527ee2ba6561f96e650878e03af25cfaf1a7f41c1dc580e61a43c9b7L44-L49)

File: android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt (hunk 2 above)
Comment (kmadsen) [reply android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt:31]: 👍 nice find! will follow up on these fixes because it will be nice to also update the examples repo

File: libnavui-androidauto/build.gradle (hunk 1 above)
Comment (kmadsen) [reply libnavui-androidauto/build.gradle]: Cool thanks for testing too, this works for me after `rm -rf ~/.gradle/caches`

File: android-auto-app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+    // Mapbox Navigation Android Auto SDK
+    implementation project(':libnavui-androidauto')
+
+    // This library will depend on the latest stable version.
+    implementation("com.mapbox.navigation:ui-dropin:2.9.0")
```
Comment (abhishek1508) [inline android-auto-app/build.gradle:73]:
```suggestion
    implementation("com.mapbox.navigation:ui-dropin:2.9.1")
```

**Review by abhishek1508 — APPROVED** [review]

File: android-auto-app/build.gradle (hunk 3 above)
Comment (kmadsen) [reply android-auto-app/build.gradle]: thanks!

File: android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt (hunk 2 above)
Comment (kmadsen) [reply android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt:31]: Ok tested this, making note of one extra step 

This works
1. Force kill app on the phone
2. ..

This reproduces the error
1. Launch app
2. Close the app
3. ...
